Implemented authorization code

emma-miler opened the pull request.

emma-miler committed:
Implemented authorization code

barnabwhy commented on sync/util.js:
```diff
@@ -4,12 +4,32 @@ const util = require('util');
 const dns = require('dns');
 const lookup = util.promisify(dns.lookup);
 
+const { WebSocket, WebSocketServer } = require('ws');
```
Is WebSocketServer needed here?

barnabwhy commented on rsa_4096_priv.pem:
```diff
@@ -0,0 +1,51 @@
+-----BEGIN RSA PRIVATE KEY-----
```
.gitignore this file

barnabwhy commented on rsa_4096_pub.pem:
```diff
@@ -0,0 +1,14 @@
+-----BEGIN PUBLIC KEY-----
```
.gitignore this file

emma-miler committed:
Performed requested changes

emma-miler committed:
Update README.md

barnabwhy committed:
ws.id in connectTo

emma-miler committed:
Update socket.js

barnabwhy reviewed (approved):
Looks good, I’d say we hold off on merging until we’ve got it all working though.

barnabwhy committed:
Remove unnecessary and somewhat breaking variable

barnabwhy committed:
Everything except actually syncing

Should be a matter of just putting in the calls to get sync data and checking they were successful

barnabwhy committed:
Disable logging (oops)

barnabwhy committed:
Update instances.example.json to have no passwords

barnabwhy committed:
Add wss:// support

barnabwhy committed:
Working data sync

emma-miler committed:
Sync network data on init

barnabwhy committed:
Add sync state check before data share

barnabwhy committed:
Merge branch 'ms-sync-websocket' of https://github.com/emma-miler/NorthstarMasterServer into ms-sync-websocket

barnabwhy committed:
Unbreak it

barnabwhy committed:
Fix data share events (!= instead of ==, oops)

barnabwhy committed:
Move to network.js as tokens.js is now redundant

barnabwhy committed:
Actually finish removing tokens

barnabwhy committed:
More descriptive

emma-miler committed:
Added network propagation

emma-miler committed:
added basic documentation, more to follow

emma-miler committed:
Create initiation.md

barnabwhy committed:
Proper SSL support and add instance data to network nodes

barnabwhy committed:
logging thing

barnabwhy committed:
add incomingAddress to example instances

barnabwhy committed:
Merge remote-tracking branch 'upstream/ms-sync-websocket' into ms-sync-websocket
